src/tests/unit/platform/test_alertsv7_v6_compatibility.py

```diff
@@ -0,0 +1,257 @@
+            check_dict(v6_sdk_response.get(key), alert_v6_from_v7.get(key), key, v6_sdk_response.get("type"))
+        else:
+            # send the dict containing the field as the field will not always exist in alert_v6_from_v7
+            check_field(v6_sdk_response, alert_v6_from_v7, key, v6_sdk_response.get("type"))
+
+
+def check_dict(alert_v6, alert_v6_from_v7, key, alert_type):
+    """
+    Make some generic checks for fields
+
+    Key should be the label, and each of alert_v6 and alert_v6_from_v7 should be dicts
+    """
+    if key in SKIP_FIELDS:
+        print("Handling known bug.  Field {}".format(key))
+        return
+    if key in COMPLEX_MAPPING_V6:
+        print("Complex mapping, ignore.  Field {}".format(key))
+        return
+    # Fields that are deprecated will be in v6 and should not be in v7
+    assert not (
+        key in BASE_FIELDS_V6 and key in alert_v6_from_v7
+    ), ("ERROR: Field is deprecated and does not exist in v7. Expected: Not in to_json(v6). Actual: was incorrectly "
+        "included. Source: BASE_FIELDS_V6. Key: {}").format(key)
+
+    # fields from cb analytics that are not in v7. No mapping available
+    assert not (
+        alert_type == "CB_ANALYTICS" and key in CB_ANALYTICS_FIELDS_V6 and key in alert_v6_from_v7
+    ), ("ERROR: Field is deprecated and does not exist in v7. Expected: Not in to_json(v6). Actual: was incorrectly "
+        "included. Source: CB_ANALYTICS_FIELDS_V6. Key: {}").format(key)
+
+    # fields from container runtime that are not in v7. No mapping available
+    assert not (alert_type == "CONTAINER_RUNTIME" and key in CONTAINER_RUNTIME_FIELDS_V6 and key in alert_v6_from_v7), (
+        ("ERROR: Field is deprecated and does not exist in v7. Expected: Not in to_json(v6). Actual: was incorrectly "
+         "included. Source: CONTAINER_RUNTIME_FIELDS_V6. Key: {}").format(key))
+
+    # no fields removed in v7 for host based firewall alerts
+    # fields from device control that are not in v7. No mapping available
+    assert not (
+        alert_type == "DEVICE_CONTROL" and key in DEVICE_CONTROL_FIELDS_V6 and key in alert_v6_from_v7
+    ), ("ERROR: Field is deprecated and does not exist in v7. Expected: Not in to_json(v6). Actual: was incorrectly "
+        "included. Source: DEVICE_CONTROL_FIELDS_V6. Key: {}").format(
+        key)
+
+    # fields from watchlist alert that are not in v7. No mapping available
+    assert not (
+        alert_type == "WATCHLIST" and key in WATCHLIST_FIELDS_V6 and key in alert_v6_from_v7
+    ), ("ERROR: Field is deprecated and does not exist in v7. Expected: Not in to_json(v6). Actual: was incorrectly "
+        "included. Source: WATCHLIST_FIELDS_V6. Key: {}").format(key)
+
+    # check dict recursively
+    # defensively coded.  This method should only be called with dicts, but avoid crashing
+    assert(isinstance(alert_v6, dict)), "Function check_dict called with incorrect argument types"
```

Comment:
This should probably be at the top of the function

Reply:
moved